### PR TITLE
fix(repo): Repository settings have been updated to accomodate deb key change

### PR DESCRIPTION
**Pull request by waynegemmell**
### PR progress checklist (to be filled in by reviewers)

- [ ] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [ ] `[feat]`     A new feature
- [x] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [ ] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No.

### Related issues and/or pull requests

None

### Describe the changes you're proposing

Debian related distributions have depricated apt-key. This change implements the aptkey = False flag in newer versions of Ubuntu/Debian. 
The repo is also cleared by setting the clean_file option to True

### Pillar / config required to test the proposed changes

It's a fundamental change so any config will trigger it. Here's an example

powerdns:
    config:
        security-poll-suffix: 
        setgid: pdns
        setuid: pdns
        webserver-address: 0.0.0.0
        webserver-allow-from: 10.0.0.0/8
        webserver-port: 8083

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### CODE REVIEW

**Comment by waynegemmell** [comment]
I see there was an issue with the commit name. I've edited it. Not sure how to re-run the tests.

File: powerdns/repo.sls [hunk 1]
```diff
@@ -4,13 +4,25 @@ powerdns_server_repo:
   pkgrepo.managed:
     - humanname: PowerDNS
     {% if salt['grains.get']('os_family') == 'Debian' %}
+    {% if (salt['grains.get']('osfullname') == "Ubuntu" and salt['grains.get']('osmajorrelease') >= 22)
+    or (salt['grains.get']('osfullname') == "Debian" and salt['grains.get']('osmajorrelease') >= 12) %}
```
Comment (vveliev-tc) [inline powerdns/repo.sls:8]: I think this logic would be better fitted in map data https://github.com/saltstack-formulas/powerdns-formula/blob/master/powerdns/osfingermap.yaml#L20

Comment (waynegemmell) [reply powerdns/repo.sls:8]: The downside is that we'll have to keep that updated for every single release.

**Comment by waynegemmell** [comment]
@vveliev-tc I've done the update. Is that right?

File: powerdns/osfingermap.yaml [hunk 2]
```diff
@@ -11,13 +11,32 @@
 # osfingermap: {}
 ---
 # os: Debian
-Debian-10: {}
-Debian-9: {}
-Debian-8: {}
+Debian-12: {
+  repo_aptkey: False
+}
```
Comment (vveliev-tc) [inline powerdns/osfingermap.yaml:16]:
```suggestion
Debian-12: 
  repo_aptkey: False

```

File: powerdns/repo.sls (hunk 1 above)
Comment (vveliev-tc) [reply powerdns/repo.sls:8]: I think the idea is to have the ability to overwrite based on the supported platforms.

you can also keep jinja yaml maps as well https://github.com/saltstack-formulas/salt-formula/blob/master/salt/osmap.yaml#L5-L16

File: powerdns/osfingermap.yaml (hunk 2 above)
Comment (vveliev-tc) [reply powerdns/osfingermap.yaml:16]: Looks good, no need for extra braces

**Comment by waynegemmell** [comment]
It would be nice to move all that logic to the yaml files but I'm not sure how much complexity they support.

File: powerdns/osfingermap.yaml (hunk 2 above)
Comment (waynegemmell) [reply powerdns/osfingermap.yaml]: I've removed them

**Comment by waynegemmell** [comment]
Anything else to be done to get this committed? @sticky-note ?

**Comment by sticky-note** [comment]
@waynegemmell  Seems legit.

**Comment by sticky-note** [comment]
@waynegemmell  Please rebase your commits and reword it to align with commitlint

**Review by sticky-note — COMMENTED** [review]
Please rebase your commit and reword it/them to align with `commitlint`

**Comment by waynegemmell** [comment]
Updated. Hopefully that's ok now.

**Comment by sticky-note** [comment]
Thanks, just check few erros on commitlint job, please
https://github.com/saltstack-formulas/powerdns-formula/pull/25/checks?check_run_id=21221085532

**Comment by sticky-note** [comment]
Hi @waynegemmell, sorry for the delay:
Would you mind to fix this problems in order to be able to merge this PR please
```
⧗   input: fix(repo): Repo settings have been updated to handle key handling change.
✖   header must not be longer than 72 characters, current length is 73 [header-max-length]
✖   subject must not be sentence-case, start-case, pascal-case, upper-case [subject-case]
✖   subject may not end with full stop [subject-full-stop]
```

**Comment by waynegemmell** [comment]
Updated. Not sure why it's failing now. It doesn't look related to my code.

**Comment by sticky-note** [comment]
Thanks for your contribution 🥇

**Comment by sticky-note** [comment]
@waynegemmell, We just released https://github.com/saltstack-formulas/powerdns-formula/releases/tag/v0.3.6.
